service/turn.rest/src/main/java/org/kaazing/gateway/service/turn/rest/TurnRestService.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.service.turn.rest;
+
+
+import java.security.KeyStore;
+import java.util.Properties;
+
+import javax.annotation.Resource;
+
+import org.apache.mina.core.session.IoSession;
+import org.kaazing.gateway.security.SecurityContext;
+import org.kaazing.gateway.service.Service;
+import org.kaazing.gateway.service.ServiceContext;
+import org.kaazing.gateway.service.ServiceProperties;
+import org.kaazing.gateway.util.feature.EarlyAccessFeatures;
+import static org.kaazing.gateway.service.util.ServiceUtils.LIST_SEPARATOR;
+
+/**
+ * Gateway service of type "turn.rest".
+ */
+public class TurnRestService implements Service {
+    
+    private static final String CLASS_PREFIX = "class:";
+    
+    private TurnRestServiceHandler handler;
+    private ServiceContext serviceContext;
+    private SecurityContext securityContext;
+    private Properties configuration;
+
+    @Resource(name = "securityContext")
+    public void setSecurityContext(SecurityContext securityContext) {
+        this.securityContext = securityContext;
+    }
+
+    @Override
+    public String getType() {
+        return "turn.rest";
+    }
+
+    @Override
+    public void init(ServiceContext serviceContext) throws Exception {
+        this.serviceContext = serviceContext;
+        ServiceProperties properties = serviceContext.getProperties();
+        KeyStore keystore = securityContext.getKeyStore();
+        
+        this.configuration = new Properties();
+        String propertyName = EarlyAccessFeatures.TURN_REST_SERVICE.getPropertyName();
+        this.configuration.setProperty(propertyName, properties.get(propertyName));
+        EarlyAccessFeatures.TURN_REST_SERVICE.assertEnabled(configuration, serviceContext.getLogger());
+        
+        ServiceProperties options = properties.getNested("options").get(0);
+        
+        StringBuilder u = new StringBuilder();
+        for (String uri: properties.getNested("uris").get(0).get("uri").split(LIST_SEPARATOR)) {
+            u.append("\"").append(uri).append("\",");
+        }
+        u.setLength(u.length() - 1);
+        String uris = u.toString();
+        
+        String credentialGeneratorClassName = properties.get("generate.credentials");
+        TurnRestCredentialsGenerator credentialGeneratorInstance;
+        if (credentialGeneratorClassName == null) {
+            return;
```

Comment:
This was the previous implementation. I simply changed the flow so that it is much more readable, but I see your point. I shall throw a ConfigurationException.